src/component/Clock/Clock.tsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+
+type ClockType = {
+  clockname:string;
+};
+
+type State = {
+  today: Date;
+};
+
+export class Clock extends React.Component<ClockType, State> {
+  state:State = {
+    today: new Date(),
+  };
+
+  upDate: NodeJS.Timer | null = null;
```

Comment:
Use `number` type instead of `NodeJS.Timer` for the `upDate` variable, as `setInterval` returns a number in the browser environment.